src/frontend/src/handler/create_sql_function.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use itertools::Itertools;
+use pgwire::pg_response::StatementType;
+use risingwave_common::catalog::FunctionId;
+use risingwave_common::types::DataType;
+use risingwave_pb::catalog::function::{Kind, ScalarFunction, TableFunction};
+use risingwave_pb::catalog::Function;
+use risingwave_sqlparser::ast::{
+    CreateFunctionBody, FunctionDefinition, ObjectName, OperateFunctionArg,
+};
+use risingwave_sqlparser::parser::Parser;
+
+use super::*;
+use crate::catalog::CatalogError;
+use crate::{bind_data_type, Binder};
+
+/// Simple type check for sql udf
+/// e.g., return type mismatch check, parameter type check, ..., etc.
+/// TODO: any existing module / function that can achieve the same functionality?
+fn _sql_udf_type_checker(ast: Vec<Statement>) -> Result<()> {
+    // Extract the expression out
+    let Statement::Query(query) = ast
+        .into_iter()
+        .exactly_one()
+        .expect("sql udf should contain only one statement")
+    else {
+        unreachable!("sql udf should contain a query statement");
+    };
+    let SetExpr::Select(select) = query.body else {
+        panic!("Invalid syntax");
+    };
+    let projection = select.projection;
+    let SelectItem::UnnamedExpr(_expr) = projection
+        .into_iter()
+        .exactly_one()
+        .expect("`projection` should contain only one `SelectItem`")
+    else {
+        unreachable!("`projection` should contain only one `SelectItem`");
+    };
+    todo!()
+}
+
+pub async fn handle_create_sql_function(
+    handler_args: HandlerArgs,
+    or_replace: bool,
+    temporary: bool,
+    name: ObjectName,
+    args: Option<Vec<OperateFunctionArg>>,
+    returns: Option<CreateFunctionReturns>,
+    params: CreateFunctionBody,
+) -> Result<RwPgResponse> {
+    if or_replace {
+        bail_not_implemented!("CREATE OR REPLACE FUNCTION");
+    }
+
+    if temporary {
+        bail_not_implemented!("CREATE TEMPORARY FUNCTION");
+    }
+
+    let language = "sql".to_string();
+    // Just a basic sanity check for language
+    if let Some(lang) = params.language {
+        let lang = lang.real_value().to_lowercase();
+        debug_assert!(lang == "sql", "`language` for sql udf function must be sql");
+    }
```

Comment:
Looks like `language` must be `sql` for SQL UDF. We can return an error if it is not. (don't panic)

Reply:
Yeah I think this is ensured in `Statement::CreateFunction`, but just in case (and for better readability), I'll add the return error statement if the `language` is not `sql`.